### DISCUSSION TITLE
Unable to read custom event type definitions from Siemens WinCC OPC UA server

### DISCUSSION THREAD

**Discussion by ts4iot** [FreeOpcUa/opcua-asyncio#1164] [Q&A]
I try to subscribe to WinCC events with the following script:

```
import asyncio
import logging
from asyncua import Client
from asyncua import ua

logging.basicConfig(level=logging.INFO)
_logger = logging.getLogger(__name__)


class SubHandler:
    """
    Subscription Handler. To receive events from server for a subscription
    data_change and event methods are called directly from receiving thread.
    Do not do expensive, slow or network operation there. Create another
    thread if you need to do such a thing
    """
    def event_notification(self, event):
        _logger.info("New event received: %r", event)
		
async def main():
	url = "opc.tcp://localhost:14862/"
	async with Client(url=url) as client:
        	# Client has a few methods to get proxy to UA nodes that should always be in address space such as Root or Objects
		_logger.info("Objects node is: %r", client.nodes.root)

        	# Now getting a variable node using its browse path
		obj = await client.nodes.root.get_child(['0:Objects', '0:Server'])
		_logger.info("MyObject is: %r", obj)

		logging.getLogger().setLevel(logging.DEBUG)
		await client.load_enums()
		await client.load_type_definitions()
		#await client.load_type_data_definitions()
		logging.getLogger().setLevel(logging.INFO)
		#x=ua.WinCCEventType()

		myevent = await client.nodes.root.get_child(['0:Types', '0:ObjectTypes', '0:BaseObjectType', '0:BaseEventType', '2:WinCCEventType'])
		_logger.info("MyFirstEventType is: %r", myevent)

		msclt = SubHandler()
		sub = await client.create_subscription(100, msclt)
		handle = await sub.subscribe_events()
		while True:
			await asyncio.sleep(1)
		await sub.unsubscribe(handle)
		await sub.delete()


if __name__ == "__main__":
    asyncio.run(main())
```

I'm able to subscribe and receive the relevant events, but they don't have the correct event type (custom fields are not available)

I try to register the custom types with:
		await client.load_enums()
		await client.load_type_definitions()
and:
		await client.load_type_data_definitions()

but without success. 
With UaExpert it is possible to read the custom fields of WinCCEventType.

Any idea how to solve the problem?

**Comment by Koseng** [comment]
Hi,
you don't give any parameters to sub.subscribe_events(). The first two parameters of subscribe_events are important. 
- sourcenode -> the node where you subscribe for the events
- evtypes -> the event types you are looking for.

If you don't give any parameters sourcenode defaults to the Server-Node and evtypes default to BaseEventType.

I think, since you only asked for the BaseEventType, you only get the BaseEventType parameters.

So your call should be like:
`handle = await sub.subscribe_events(evtypes=myevent)`

**Comment by ts4iot** [comment]
Hi,
thank you for your support, it helps me to solve my problem. When I changed the code according to your suggestion, I received no event at all. This shows me that the event I received before was not of type:

```['0:Types', '0:ObjectTypes', '0:BaseObjectType', '0:BaseEventType', '2:WinCCEventType']```

it is of type:

```['0:Types', '0:ObjectTypes', '0:BaseObjectType', '0:BaseEventType', '0:ConditionType', '0:AcknowledgeableConditionType', '0:AlarmConditionType', '2:WinCCAlarmConditionType']```
 
Now the event object has all the field as expected.

**Reply by Koseng** [reply]
If you are interested in Alarms and Conditions use `sub.subscribe_alarms_and_conditions(....)`

See example:
https://github.com/FreeOpcUa/opcua-asyncio/blob/master/examples/client_to_uanet_alarm_conditions.py
